GitUI/GitExtensionsForm.cs

```diff
@@ -132,14 +132,16 @@ protected virtual void RestorePosition()
                     location.Y = rect.Y;
                 }
 
+                location.X = Math.Max(0, location.X);
+                location.Y = Math.Max(0, location.Y);
                 DesktopLocation = location;
             }
             else
             {
                 // Calculate location for modal form with parent
                 Location = new Point(
-                    Owner.Left + (Owner.Width / 2) - (Width / 2),
-                    Owner.Top + (Owner.Height / 2) - (Height / 2));
+                    Math.Max(0, Owner.Left + (Owner.Width / 2) - (Width / 2)),
+                    Math.Max(0, Owner.Top + (Owner.Height / 2) - (Height / 2)));
```

Comment:
Unfortunately, this will not work. Windows screen coordinates _can_ be negative, e.g. if you arrange a second display left of the primary display.
I guess we must iterate all displays and check against their display areas (excluding the taskbar).

Reply:
Thank you. Yes, I realised that, I just don't have a 2nd monitor to test. I should get one soon, and I'd be able to resume this work.

Reply:
There is quite a lot of code checking `Screen.WorkingArea`s in `GitExtensionsForm` and `WindowPositionManager` already.
It looks sensible. Though it is done before the DPI-scaling.
Often, the union of the `WorkingAreas` will be no rectangle.
A pragmatic approach could be to constrain the Form to the `WorkingArea` which contains the center or the maximum part of the Form.